usefulUtilities/airAndWhiteboardDraw/entityResources/entityClientScripts/drawSphereClient.js

```diff
@@ -0,0 +1,832 @@
+    var DEFAULT_NORMAL = { x: 0, y: 0, z: 1 };
+    var DECAY_TIME_S = 60;
+    var MAX_LINE_POINTS = 100;
+    var DRAW_SOUND = SoundCache.getSound(Script.resolvePath('../resources/sounds/draw.mp3'));
+    var DRAW_SOUND_VOLUME = 0.08;
+    var DELETE_AGAIN_MS = 100;
+    var MAXIMUM_DISTANCE_TO_SEARCH_M = 1;
+    var MAXIMUM_DISTANCE_TO_DELETE_M = 0.03;
+    var DISTANCE_TO_DRAW_IN_FRONT_OF_CAMERA_DESKTOP_M = 1.5;
+    var DRAW_ON_BOARD_DISTANCE_HMD_M = 0.01;
+    var DRAW_ON_BOARD_DISTANCE_DESKTOP_M = 3;
+    var OPEN_SOUND = SoundCache.getSound(Script.resolvePath('../resources/sounds/open.mp3'));
+    var OPEN_SOUND_VOLUME = 0.02;
+    var CLOSE_SOUND = SoundCache.getSound(Script.resolvePath('../resources/sounds/close.mp3'));
+    var CLOSE_SOUND_VOLUME = 0.02;
+    var WAIT_TO_CLEAN_UP_MS = 2000;
+    var WAIT_TO_REOPEN_APP_MS = 500;
+    var WAIT_FOR_ENTITIES_TO_LOAD_MS = 300;
+    var STROKE_FORWARD_OFFSET_M = 0.01;
+    var DEFAULT_LINE_PROPERTIES = {
+        type: "PolyLine",
+        name: "Whiteboard Polyline",
+        isUVModeStretch: true,
+        lifetime: DECAY_TIME_S,
+        collisionless: true,
+        grab: { grabbable: false }
+    };
+    var HALF = 0.5;
+
+    var isPainting = false;
+    var readyToDraw = false;
+    var isMouseDown = false;
+    var isStartingStroke = false;
+    var lastIntersectionPoint = undefined;
+    var lineResolution = 0.01;
+    var timestamp = null;
+    var strokeSoundTimestamp0 = null, strokeSoundTimestamp1 = null;
+    var cursorID = undefined;
+    var mouseMoveTimestamp = Date.now();
+    var hmdMoveTimestamp = Date.now();
+    var hapticPulseTimestamp = Date.now();
+    var equipTimestamp = undefined;
+    var dominantHandJoint;
+    var dominantHand;
+    var tablet = Tablet.getTablet('com.highfidelity.interface.tablet.system');
+    var whiteboard = null;
+    // Performance Debug
+    // 2 / 60.0 *1000
+    // 1/ 60.0*1000
+    var throttleTimeoutMS = 16.6;
+    var hapticTimeoutMS = 140;
+    var controllerMapping;
+    var distanceCheckInterval = null;
+    var deletingInterval;
+    var activeTriggerPress = false;
+    var activeGripPress = false;
+    var controllerMappingName = 'Hifi-DrawApp';
+    var animationData = {};
+    var animationHandlerID;
+    var injector;
+    var mouseEventsConnected = false;
+    var paintSphereDimensions;
+    var linePoints = [{x: 0, y: 0, z: 0 }];
+    var lineNormals = [DEFAULT_NORMAL, DEFAULT_NORMAL];
+    var lineStrokeWidths = [];
+    var lineStartPosition;
+    var pickRay;
+    var desktopActionInProgress = false;
+    var polyLine = null;
+    var previousLinePoint;
+    var previousNormal;
+    var previousStrokeWidth;
+    var currentStrokeWidth;
+    var parentJointIndex;
+    var currentLinePoint;
+    var wasLastPointOnBoard;
+    var whiteboardParts = [];
+
+    var PaintSphere = function() {
+        _this = this;
+    };
+
+    PaintSphere.prototype = {
+        preload: function(entityID) {
```

Comment:
add comments above all functions (i'll only make this comment one time to reduce clutter)